### ISSUE TITLE
numpy 1.11.0 cannot be installed with pip-accel

### ISSUE THREAD

**Issue by kinghajj** [numpy/numpy#7632]
```
[python] 2016-05-13 20:46:42 ef3f46b87f25 pip_accel[119] INFO Installing numpy (1.11.0) binary distribution using pip-accel ..
[python] 2016-05-13 20:46:42 ef3f46b87f25 pip_accel.bdist[119] INFO Building numpy (1.11.0) binary distribution ..
[python] 2016-05-13 20:46:43 ef3f46b87f25 pip_accel.bdist[119] WARNING Build of numpy (1.11.0) failed, falling back to alternative method ..
[python] 2016-05-13 20:46:43 ef3f46b87f25 pip_accel.bdist[119] INFO Building numpy (1.11.0) binary distribution ..
[python] 2016-05-13 20:46:43 ef3f46b87f25 pip_accel.bdist[119] WARNING Build of numpy (1.11.0) failed, checking for missing dependencies ..
[python] 2016-05-13 20:46:43 ef3f46b87f25 pip_accel.deps[119] INFO Checking for known dependencies of numpy ..
[python] 2016-05-13 20:46:43 ef3f46b87f25 pip_accel.deps[119] INFO No known dependencies... Maybe you have a suggestion?
[python] 2016-05-13 20:46:43 ef3f46b87f25 pip_accel.cli[119] ERROR Caught unhandled exception!
[python] Traceback (most recent call last):
[python]   File "/build/venv/lib/python2.7/site-packages/pip_accel/cli.py", line 57, in main
[python]     accelerator.install_from_arguments(arguments)
[python]   File "/build/venv/lib/python2.7/site-packages/pip_accel/__init__.py", line 246, in install_from_arguments
[python]     return self.install_requirements(requirements, **kw)
[python]   File "/build/venv/lib/python2.7/site-packages/pip_accel/__init__.py", line 584, in install_requirements
[python]     self.bdists.install_binary_dist(binary_distribution, **kw)
[python]   File "/build/venv/lib/python2.7/site-packages/pip_accel/bdist.py", line 437, in install_binary_dist
[python]     for member, from_handle in members:
[python]   File "/build/venv/lib/python2.7/site-packages/pip_accel/bdist.py", line 87, in get_binary_dist
[python]     raw_file = self.build_binary_dist(requirement)
[python]   File "/build/venv/lib/python2.7/site-packages/pip_accel/bdist.py", line 217, in build_binary_dist
[python]     return self.build_binary_dist_helper(requirement, ['bdist', '--formats=gztar'])
[python]   File "/build/venv/lib/python2.7/site-packages/pip_accel/bdist.py", line 282, in build_binary_dist_helper
[python]     name=requirement.name, version=requirement.version)
[python] BuildFailed: Failed to build numpy (1.11.0) binary distribution!
[python] 
[python] Please check the build output because it will probably provide a hint about what went wrong.
[python] 
[python] Build output:
[python] 
[python] Running from numpy source directory.
[python] `setup.py bdist` is not supported
[python] Add `--force` to your command to use it anyway if you must (unsupported).
```

Is there a suggested way around this, or is it an oversight/bug?

Edit: 1.10.4 still builds fine.

**Comment by njsmith** [comment]
It would probably be easy to hack this on numpy's end to make it mostly-work for a while longer, but IMO the real fix is for pip-accel to attempt building a wheel (ideally with `pip wheel ...`) before falling back to `setup.py bdist_dumb` or `setup.py bdist`. As part of the effort to enable non-distutils build systems, the upstream python packaging world is moving towards standardizing on wheel-building as the *only* build operation that all python sdists are guaranteed to support -- see recent threads on distutils-sig.

Could you file a bug with pip-accel to see what they think about this?

**Comment by njsmith** [comment]
(I think we'd also accept a PR to add bdist_dumb to the whitelist of allowed commands in numpy's setup.py, but you should file that bug in any case :-))

**Comment by rgommers** [comment]
I'm not sure we should allow either ``bdist`` or ``bdist_dumb``. If you want to generate a source distribution via setup.py, then ``sdist`` is the right command. And we add a check (https://github.com/numpy/numpy/blob/master/setup.py#L196) to prevent missing files that ``bdist`` and ``bdist_dumb`` don't have. And if the usecase is generating another source dist from a downloaded sdist, that's just weird....

**Comment by rgommers** [comment]
I've also never seen ``bdist_dumb`` used before, there is basically no documentation for it, and a Google search finds on the first page a [blog post](http://lucumr.pocoo.org/2014/1/27/python-on-wheels/) by Armin Ronacher which only says "historically bdist_dumb was broken".

**Comment by njsmith** [comment]
I don't really understand bdist_dumb either, but I think it shares more code paths with bdist_wheel than it does with sdist. Afaict it's a kind of mostly unspecified Proto-egg/wheel format

**Comment by rgommers** [comment]
Right, I was misled by the few lines of description I found online. The code for ``bdist_dumb`` is actually pretty clean for a distutils command.

**Comment by rgommers** [comment]
Still, re-enabling support for stuff that distutils-sig has given up on ages ago would feel wrong.....

**Comment by njsmith** [comment]
Yeah, my feeling is that OK, if we broke pip-accel that's unfortunate, so if someone does the (probably very minor) work to make it keep working temporarily then I'm OK with that... as long as it's understood that this is not a configuration that we support ourselves and as soon as it starts generating support requests or interferes with making other stuff work then out it goes?

**Comment by rgommers** [comment]
sounds fair

**Comment by vidstige** [comment]
What's the latest development on this? It would be fantastic to be able to use py2deb to package create fresh versions of numpy instead of running on superold versions.

**Comment by mahmuuud** [comment]
Hello @njsmith Is there any update on this? I am facing a production issue with it, do you have any suggestions on how to unblock and use Numpy with py2deb?

I tried adding bdist to the whitelist of commands and using my own version of Numpy but it still fails due to the same error, don't know if I am missing something here, Your help would be appreciated.

**Comment by rgommers** [comment]
We have now switched build systems from `numpy.distutils` to Meson, so this issue is no longer relevant (there's are no `bdist_*` commands anymore). Hence closing - thanks all.
